python/cortex.py

```diff
@@ -27,16 +29,353 @@
+                self.setup_profile(self.profile_name, 'load')
+            else:
+                loaded_by_this_app = result_dic['loadedByThisApp']
+                print('get current profile rsp: ' + name + ", loadedByThisApp: " + str(loaded_by_this_app))
+                if name != self.profile_name:
+                    warnings.warn("There is profile " + name + " is loaded for headset " + self.headset_id)
+                elif loaded_by_this_app == True:
+                    self.emit('load_unload_profile_done', isLoaded=True)
+                else:
+                    self.setup_profile(self.profile_name, 'unload')
+                    # warnings.warn("The profile " + name + " is loaded by other applications")
+        elif req_id == DISCONNECT_HEADSET_ID:
+            print("Disconnect headset " + self.headset_id)
+            self.headset_id = ''
+        elif req_id == MENTAL_COMMAND_ACTIVE_ACTION_ID:
+            self.emit('get_mc_active_action_done', data=result_dic)
+        elif req_id == MENTAL_COMMAND_TRAINING_THRESHOLD:
+            self.emit('mc_training_threshold_done', data=result_dic)
+        elif req_id == MENTAL_COMMAND_BRAIN_MAP_ID:
+            self.emit('mc_brainmap_done', data=result_dic)
+        elif req_id == SENSITIVITY_REQUEST_ID:
+            self.emit('mc_action_sensitivity_done', data=result_dic)
+        elif req_id == CREATE_RECORD_REQUEST_ID:
+            self.record_id = result_dic['record']['uuid']
+            self.emit('create_record_done', data=result_dic['record'])
+        elif req_id == STOP_RECORD_REQUEST_ID:
+            self.emit('stop_record_done', data=result_dic['record'])
+        elif req_id == EXPORT_RECORD_ID:
+            # handle data lable
+            success_export = []
+            for record in result_dic['success']:
+                record_id = record['recordId']
+                success_export.append(record_id)
+
+            for record in result_dic['failure']:
+                record_id = record['recordId']
+                failure_msg = record['message']
+                print('export_record resp failure cases: '+ record_id + ":" + failure_msg)
+
+            self.emit('export_record_done', data=success_export)
+        elif req_id == INJECT_MARKER_REQUEST_ID:
+            self.emit('inject_marker_done', data=result_dic['marker'])
+        elif req_id == INJECT_MARKER_REQUEST_ID:
+            self.emit('update_marker_done', data=result_dic['marker'])
+        else:
+            print('No handling for response of request ' + req_id)
+
+    def handle_error(self, recv_dic):
+        req_id = recv_dic['id']
+        error_dic = recv_dic['error']
+        error_code = error_dic['code']
+
+        if error_code == ERR_PROFILE_ACCESS_DENIED:
+            #disconnect headset
+            error_message = error_dic['message']
+            warnings.warn(error_message)
+
+            self.disconnect_headset()
+        else:
+            print(recv_dic)
```

Comment:
Why there's no disconnect_headset call for the other types of error? @tungntEmotiv

Reply:
Print the error to the console would not allow end-user to run their custom error handler, better to raise an explicit exception here with error result

Reply:
I will emit error to main script to handle. Almost cases , will raise a exception to terminate the program and do not need disconnect headset because it might impact to other applications.  Only the case errorcode=-32046, we need to call disconnect to force unload profile for next use.

Reply:
I have updated rework , @thamha-emotiv , @NguyenDuyToan , a @tanpv , Please help to review again. Thanks